PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -359,46 +475,26 @@ struct lambdaCorrTableProducer {
       auto postrack = v0.template posTrack_as<Tracks>();
       auto negtrack = v0.template negTrack_as<Tracks>();
 
-      // get lambda
-      if ((fabs(v0.mLambda() - MassLambda0) < cfg_lambda_mass_window) && (fabs(v0.mK0Short() - MassK0Short) > cfg_kshort_rej) && (selPIDTrack(postrack, postrack.tpcNSigmaPr(), postrack.tofNSigmaPr())) && (selPIDTrack(negtrack, negtrack.tpcNSigmaPi(), negtrack.tofNSigmaPi()))) {
-        fillQALambda<kLambda>(collision, v0, tracks);
-        lambdaTrackTable(lambdaCollisionTable.lastIndex(), v0.pt(), v0.yLambda(), v0.eta(), v0.phi(), v0.mLambda(), postrack.index(), negtrack.index(), true);
-      }
-
-      // get anti-lambda
-      if ((fabs(v0.mAntiLambda() - MassLambda0) < cfg_lambda_mass_window) && (fabs(v0.mK0Short() - MassK0Short) > cfg_kshort_rej) && (selPIDTrack(negtrack, negtrack.tpcNSigmaPr(), negtrack.tofNSigmaPr())) && (selPIDTrack(postrack, postrack.tpcNSigmaPi(), postrack.tofNSigmaPi()))) {
-        fillQALambda<kAntiLambda>(collision, v0, tracks);
-        lambdaTrackTable(lambdaCollisionTable.lastIndex(), v0.pt(), v0.yLambda(), v0.eta(), v0.phi(), v0.mAntiLambda(), postrack.index(), negtrack.index(), false);
-      }
+      selV0Particle<kLambda, kProton, kPion, kCentralWindow>(collision, v0, postrack, negtrack, tracks);
+      selV0Particle<kLambda, kProton, kPion, kLeftWindow>(collision, v0, postrack, negtrack, tracks);
+      selV0Particle<kLambda, kProton, kPion, kRightWindow>(collision, v0, postrack, negtrack, tracks);
+      selV0Particle<kAntiLambda, kPion, kProton, kCentralWindow>(collision, v0, postrack, negtrack, tracks);
+      selV0Particle<kAntiLambda, kPion, kProton, kLeftWindow>(collision, v0, postrack, negtrack, tracks);
+      selV0Particle<kAntiLambda, kPion, kProton, kRightWindow>(collision, v0, postrack, negtrack, tracks);
```

Comment:
In here for the same v0 each of the possibilities is tried although only one will succeed (I guess they are mutually exclusive)
For the next iteration I would suggest simplify this a bit. Most probably the extraction of the invariant mass window should be immediate
Also, as the tracks table is passed it is not needed to additionally pass `postrack` and `negtrack